src/modules/ztsi/src/lib/Ztsi.cpp

```diff
@@ -313,40 +344,62 @@ int Ztsi::SetEnabled(bool enabled)
         if (enabled != configuration.enabled)
         {
             configuration.enabled = enabled;
-            status = IsValidConfiguration(configuration) ? WriteAgentConfiguration(configuration) : MMI_OK;
+            status = IsValidConfiguration(configuration) ? WriteAgentConfiguration(configuration) : EINVAL;
         }
     }
     else if (ENOENT == status)
     {
         // If the configuration file doesn't exist, create it with the desired enabled state
         configuration.enabled = enabled;
-        configuration.serviceUrl = g_defaultServiceUrl;
-        status = IsValidConfiguration(configuration) ? CreateConfigurationFile(configuration) : MMI_OK;
+        status = IsValidConfiguration(configuration) ? CreateConfigurationFile(configuration) : EINVAL;
     }
 
     return status;
 }
 
-int Ztsi::SetServiceUrl(const std::string& serviceUrl)
+int Ztsi::SetMaxScheduledAttestationsPerDay(int maxScheduledAttestationsPerDay)
 {
     int status = MMI_OK;
-    AgentConfiguration configuration = {g_defaultServiceUrl, g_defaultEnabled};
+    AgentConfiguration configuration = {g_defaultEnabled, g_defaultMaxScheduledAttestationsPerDay, g_defaultMaxManualAttestationsPerDay};
+    status = ReadAgentConfiguration(configuration);
+    if ((MMI_OK == status) || (EINVAL == status))
+    {
+        if (maxScheduledAttestationsPerDay != configuration.maxScheduledAttestationsPerDay){
+            configuration.enabled = m_lastEnabledState;
+            configuration.maxScheduledAttestationsPerDay = maxScheduledAttestationsPerDay;
+            status = IsValidConfiguration(configuration) ? WriteAgentConfiguration(configuration) : EINVAL;
+        }
+    }
+    else if (ENOENT == status)
+    {
+        // If the configuration file doesn't exist, create it with the desired maxScheduledAttestationsPerDay state
+        configuration.enabled = m_lastEnabledState;
+        configuration.maxScheduledAttestationsPerDay = maxScheduledAttestationsPerDay;
+        status = IsValidConfiguration(configuration) ? CreateConfigurationFile(configuration) : EINVAL;
+    }
+
+    return status;
+}
+
+int Ztsi::SetMaxManualAttestationsPerDay(int maxManualAttestationsPerDay)
+{
+    int status = MMI_OK;
+    AgentConfiguration configuration = {g_defaultEnabled, g_defaultMaxScheduledAttestationsPerDay, g_defaultMaxManualAttestationsPerDay};
 
     status = ReadAgentConfiguration(configuration);
     if ((MMI_OK == status) || (EINVAL == status))
     {
-        if (serviceUrl != configuration.serviceUrl)
-        {
+        if (maxManualAttestationsPerDay != configuration.maxManualAttestationsPerDay){
             configuration.enabled = m_lastEnabledState;
-            configuration.serviceUrl = serviceUrl;
+            configuration.maxManualAttestationsPerDay = maxManualAttestationsPerDay;
             status = IsValidConfiguration(configuration) ? WriteAgentConfiguration(configuration) : EINVAL;
```

Comment:
>status = IsValidConfiguration(configuration) ? WriteAgentConfiguration(configuration) : EINVAL; [](http://example.com/codeflow?start=12&length=95)

Errors should be logged

Reply:
Wouldn't the relevant errors be logged in IsValidConfiguration()?